cmd/compute-domain-controller/cleanup.go

```diff
@@ -23,13 +23,17 @@ type CleanupManager[T metav1.Object] struct {
 	informer         cache.SharedIndexInformer
 	getComputeDomain GetComputeDomainFunc
 	callback         CleanupCallback[T]
+	queue            chan struct{}
 }
 
 func NewCleanupManager[T metav1.Object](informer cache.SharedIndexInformer, getComputeDomain GetComputeDomainFunc, callback CleanupCallback[T]) *CleanupManager[T] {
 	return &CleanupManager[T]{
 		informer:         informer,
 		getComputeDomain: getComputeDomain,
 		callback:         callback,
+		// Buffered channel to implement a pragmatic fixed-size queue so that
+		// we enqueue at most one cleanup operation.
+		queue: make(chan struct{}, 1),
```

Comment:
I like this super simple approach; thought about using pkg/workqueue and am not convinced we want that retrying+backoff behavior here.